include/oneapi/dpl/pstl/utils.h

```diff
@@ -582,6 +582,29 @@ template <typename _Tp>
 using __is_const_callable_object =
     ::std::integral_constant<bool, __is_callable_object<_Tp>::value && __is_pointer_to_const_member<_Tp>::value>;
 
+struct __next_to_last
+{
+    template <typename _Iterator>
+    typename ::std::enable_if<::std::is_same<typename ::std::iterator_traits<_Iterator>::iterator_category,
```

Comment:
`is_same` should be replaced by `is_base_of` because we have `contiguous_iterator_tag` since C++20. I don't insist that we do it right now but please do it immediately after merging this review as bug fix